compiler/rustc_ast/src/tokenstream.rs

```diff
@@ -312,9 +312,16 @@ pub struct TokenStream(pub(crate) Lrc<Vec<TokenTree>>);
 #[cfg(all(target_arch = "x86_64", target_pointer_width = "64"))]
 rustc_data_structures::static_assert_size!(TokenStream, 8);
 
+// Simlar to `proc_macro::Spacing`, but for tokens.
 #[derive(Clone, Copy, Debug, PartialEq, Encodable, Decodable, HashStable_Generic)]
 pub enum Spacing {
+    /// The token is not immediately followed by an operator token (as
+    /// determined by `Token::is_op`). E.g. a `+` token is `Alone` in `+ =`,
+    /// `+/*foo*/=`, `+ident`, and `+()`.
     Alone,
+
+    /// The token is immediately followed by an operator token. E.g. a `+`
+    /// token is `Joint` in `+=` and `++`.
     Joint,
```

Comment:
These are not pedantically correct: for things like ident, I think we still call them `Alone` even if they are joint by an operator? Eg, in `id+` `id` would be alone.

Hm, re-reading the code, I think this definition is correct (that is, `id` would be joint). But I think it would be `alone` once it gets to proc-macro layer? 

I guess in any case its worth clarifying semantics for thigs which are not themselves operators.

Reply:
Once it gets to the proc-macro layer only `Punct` token trees have a `Spacing`. Non-`Punct` ones (e.g. `Ident`) don't have a `Spacing`. So the comment is correct, but I'll add a note about the difference to the proc-macro types.